src/bokeh/util/callback_manager.py

```diff
@@ -205,13 +205,12 @@ def _nargs(fn: Callable[..., Any]) -> int:
 def _check_callback(callback: Callable[..., Any], fargs: Sequence[str], what: str ="Callback functions") -> None:
     '''Bokeh-internal function to check callback signature'''
     sig = signature(callback)
-    formatted_args = str(sig)
-    error_msg = what + " must have signature func(%s), got func%s"
-
     all_names, default_values = get_param_info(sig)
 
     nargs = len(all_names) - len(default_values)
     if nargs != len(fargs):
+        formatted_args = str(sig)
+        error_msg = what + " must have signature func(%s), got func%s"
         raise ValueError(error_msg % (", ".join(fargs), formatted_args))
```

Comment:
@Latill we have gotten rid of almost all old-style string formatting in Bokeh, but a few spots still remain, like this one. As long as you are touching this code, can you update this to use an f-string? Something like (untested)

```suggestion
        expected = ", ".join(fargs)
        received = str(sig)
        error_msg = f"{what} must have signature func{expected}, got func{received}"
        raise ValueError(error_msg)
```

Reply:
Sure, should I make a new commit, push into the same branch, and make a new pull request from scratch ? or I can update this one in some way ?

Reply:
@Latill no need for a new PR, you can just make a new commit and push into the existing branch, and this PR will update automatically.